stock_request/models/stock_request_abstract.py

```diff
@@ -238,23 +231,33 @@ def onchange_location_id(self):
                 self.warehouse_id = loc_wh
                 self.with_context(no_change_childs=True).onchange_warehouse_id()
 
+    @api.onchange("allow_virtual_location")
+    def onchange_allow_virtual_location(self):
+        if self.allow_virtual_location:
+            return {"domain": {"location_id": []}}
```

Comment:
I'm not in favor reintroducing domains in onchanges as this is deprecated in core and will be removed in later version.

Reply:
I agree here. This should be handed on a static domain, but you can use simple expressions on domains. Example: https://github.com/OCA/rma/blob/e0c17a60e152f8814242e026166e1f2114789944/rma_sale/models/rma.py#L41-L42